### PR TITLE
Pin auth to v0 and show full permissions block in examples

### PR DESCRIPTION
**Pull request by sethvargo**
Refs https://github.com/google-github-actions/setup-gcloud/issues/430

### CODE REVIEW

**Comment by sethvargo** [comment]
https://github.com/google-github-actions/deploy-cloud-functions/pull/226/files?w=1 is an easier review

File: README.md [hunk 1]
```diff
@@ -30,22 +30,30 @@ Cloud Function. See the Authorization section below for more information.
...
+    - uses: actions/checkout@v2
+
+    - id: auth
+      uses: google-github-actions/auth@v0
+      with:
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    - id: deploy
+      uses: google-github-actions/deploy-cloud-functions@v0.6.0
```
Comment (bharathkkb) [inline README.md:49]:
```suggestion
      uses: google-github-actions/deploy-cloud-functions@v0.7.0
```

File: README.md [hunk 2]
```diff
@@ -206,32 +214,46 @@ See [usage](https://github.com/google-github-actions/auth#usage) for more detail
...
```
Comment (bharathkkb) [inline README.md:233]:
```suggestion
      uses: google-github-actions/deploy-cloud-functions@v0.7.0
```

File: README.md [hunk 3]
```diff
@@ -206,32 +214,46 @@ See [usage](https://github.com/google-github-actions/auth#usage) for more detail
...
+    steps:
+    - uses: actions/checkout@v2
+
+    - id: auth
+      uses: google-github-actions/auth@v0
+      with:
+        credentials_json: ${{ secrets.gcp_credentials }}
+
+    - id: deploy
+      uses: google-github-actions/deploy-cloud-functions@v0.6.0
```
Comment (bharathkkb) [inline README.md:253]:
```suggestion
      uses: google-github-actions/deploy-cloud-functions@v0.7.0
```

File: README.md [hunk 4]
```diff
@@ -242,12 +264,16 @@ authenticate requests as the service account attached to the instance. **This
...
-  with:
-    name: my-function
-    runtime: nodejs10
+jobs:
+  job_id:
+    steps:
+    - uses: actions/checkout@v2
+
+    - id: Deploy
+      uses: google-github-actions/deploy-cloud-functions@v0.6.0
```
Comment (bharathkkb) [inline README.md:273]:
```suggestion
      uses: google-github-actions/deploy-cloud-functions@v0.7.0
```

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**